### PR TITLE
Feature/orca 244

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-244: Create schema files for copy_to_glacier](https://bugs.earthdata.nasa.gov/browse/ORCA-244)

## Changes

* Added schemas for copy_to_glacier
* Added schema tests

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Added unit tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [x] readme
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
    - [ ] Integration tests
    - [ ] Manual tests (outlined above)
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: tasks/copy_to_glacier/test/unit_tests/test_copy_to_glacier.py [hunk 1]
```diff
@@ -75,6 +78,12 @@ class TestCopyToGlacierHandler(TestCase):
         ]
     }
 
+    @patch("copy_to_glacier.task")
+    def test_handler_happy_path(self,
+                                mock_task: MagicMock):
+        # todo
```
Comment (AndrewDorn) [inline tasks/copy_to_glacier/test/unit_tests/test_copy_to_glacier.py:84]: Complete to test /input

File: tasks/copy_to_glacier/schemas/output.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://github.com/nasa/cumulus-orca/blob/master/tasks/copy_to_files/schemas/output.json",
+  "title": "copy_to_files Lambda Output",
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/output.json:4]: Shouldn't this be `copy_to_glacier` in the **id**, **title**, and **description**?

File: tasks/copy_to_glacier/schemas/output.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            "type": "string"
+          },
+          "files": {
+            "description": "The files to copy.",
+            "type": "array",
+            "items": {
+              "required": ["name", "bucket", "filepath", "filename"],
+              "properties": {
+                "name": {
+                  "type": "string"
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/output.json:26]: Should add a description to each of these properties so we know what they are and what to expect for a value.

File: tasks/copy_to_glacier/schemas/input.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://github.com/nasa/cumulus-orca/blob/master/tasks/copy_to_files/schemas/input.json",
+  "title": "copy_to_files Lambda Input",
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/input.json:4]: Shouldn't this be `copy_to_glacier` in the **id**, **title**, and **description**?

File: tasks/copy_to_glacier/schemas/input.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            "type": "string"
+          },
+          "files": {
+            "description": "The files to copy.",
+            "type": "array",
+            "items": {
+              "required": ["name", "bucket", "filepath", "filename"],
+              "properties": {
+                "name": {
+                  "type": "string"
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/input.json:26]: should add a description to each of these properties so users know what to expect for values.

File: tasks/copy_to_glacier/schemas/config.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://github.com/nasa/cumulus-orca/blob/master/tasks/copy_to_files/schemas/config.json",
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/config.json:3]: Shouldn't this be `copy_to_glacier` in the **id**, **title**, and **description**?

File: tasks/copy_to_glacier/schemas/config.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  "title": "copy_to_files Lambda Config",
+  "description": "The config for the copy_to_files Lambda.",
+  "type": "object",
+  "required": ["collection"],
+  "properties": {
+    "collection": {
+      "description": "The collection from AWS.",
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/config.json:10]: This collection is not from AWS but comes from Cumulus.

File: tasks/copy_to_glacier/schemas/config.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      "type": "object",
+      "required": ["name", "version", "files", "url_path"],
```
Comment (bhazuka) [inline tasks/copy_to_glacier/schemas/config.json:12]: Looking at the code I do not believe any of these properties are needed or required. I believe all we really need is the `meta` property object and optionally the `excludeFileTypes` property list nested within the object.

File: tasks/copy_to_glacier/test/unit_tests/test_copy_to_glacier.py [hunk 9]
```diff
@@ -75,6 +81,48 @@ class TestCopyToGlacierHandler(TestCase):
...
+                           "filepath": uuid.uuid4().__str__(), "filename": uuid.uuid4().__str__()}]
+            }
+        ]
+
+        handler_input_event = {
+            "payload": {
+                "granules": granules
+            },
+            "task_config": {
+                "collection": {
```
Comment (bhazuka) [inline tasks/copy_to_glacier/test/unit_tests/test_copy_to_glacier.py:100]: See comments on config schema. This may need to be adjusted accordingly.

**Review by bhazuka — CHANGES_REQUESTED** [review]
A few minor items to look at.

File: tasks/copy_to_glacier/README.md [hunk 10]
```diff
@@ -274,32 +246,24 @@ FUNCTIONS
         
             Environment Variables:
                 ORCA_DEFAULT_BUCKET (string, required): Name of the default ORCA S3 Glacier bucket.
-                DEFAULT_MULTIPART_CHUNKSIZE_MB (int, optional): The default maximum size of chunks to use when copying. Can be overridden by collection config.
+                DEFAULT_MULTIPART_CHUNKSIZE_MB (string, optional): The default maximum size of chunks to use when copying. Can be overridden by collection config.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/copy_to_glacier/README.md:249]: Not sure what I was going for here. Fixed.

File: tasks/copy_to_glacier/requirements-dev.txt [hunk 11]
```diff
@@ -1,4 +1,6 @@
+cumulus-message-adapter-python>=1.1.1
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/copy_to_glacier/requirements-dev.txt:1]: Think I copied this in by accident. Removed.

File: tasks/copy_to_glacier/schemas/output.json (hunk 2 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/output.json:4]: Whoops. Fixed in all files.

File: tasks/copy_to_glacier/schemas/input.json [hunk 12]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://github.com/nasa/cumulus-orca/blob/master/tasks/copy_to_files/schemas/input.json",
+  "title": "copy_to_files Lambda Input",
+  "description": "The input for the copy_to_files Lambda.",
+  "type": "object",
+  "required": ["granules"],
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/input.json:7]: Fixed

File: tasks/copy_to_glacier/schemas/config.json (hunk 7 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/config.json:10]: Fixed

File: tasks/copy_to_glacier/schemas/input.json (hunk 4 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/input.json:4]: Fixed.

File: tasks/copy_to_glacier/schemas/config.json (hunk 6 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/config.json:3]: Fixed

File: tasks/copy_to_glacier/schemas/output.json (hunk 3 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/output.json:26]: Gave it a shot.

File: tasks/copy_to_glacier/schemas/input.json (hunk 5 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/input.json:26]: Gave it a shot.

File: tasks/copy_to_glacier/schemas/config.json (hunk 8 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/config.json:12]: Hm. Looks like this got reworked at some point, and the properties weren't removed. Fixed.

Comment (AndrewDorn) [reply tasks/copy_to_glacier/schemas/config.json:12]: `meta` is optional as well.

**Review by bhazuka — APPROVED** [review]

**Pull request merged**